infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/NamespaceNameValidator.java

```diff
@@ -51,6 +52,15 @@ public static boolean isValid(String name) {
     return validateInternal(name).isOk();
   }
 
+  public static String reduceToValid(String namespaceName) {
```

Comment:
Is it OK to have such method here? Just to avoid duplicating of validation regex somewhere else..

Reply:
I'm not sure this works well.

`a-b#-hello` would be reduced to `a-bhello` (I think) which I am not sure is correct - I would have assumed `a-b-hello`.

But overall I'm not sure if removing the invalid chars is better than replacing them with some placeholder (and making sure that the name is valid after the replacement).

Reply:
You're right, but i'm not sure if we should implement a complex logic to be 100% precise, for me not big difference between in result of `a-bhello-jsb786` and `a-b-hello-jkh762`, it anyway be faintly readable. (but if we can reach that easily it would be good). About placeholders...what we can use, imo only "-"  -s ? Does it really makes sense to have
them like `a-b---hello--jkh762-` and the iterate couple of times?

Reply:
In a lot of cases, an invalid name would be a valid email. So lets imagine how `blah@gmail.com` would look better - `blahgmailcom-jfk3434` or `blah-gmail-com-jfk3434`. Anyway, `dash` on a first position has to be removed one way or another. I vote 40%-60% for the second variant. We can even put this replacement character in the configuration.
Also, this name has to be stored somewhere. In user preferences for example.